src/main/scala/scalaz/analytics/AnalyticsModule.scala

```diff
@@ -100,19 +100,33 @@ trait AnalyticsModule {
    * The DataSet/DataStream operations of scalaz-analytics
    */
   trait Ops[F[_]] {
-    // Unbounded
+    // Unbounded dataset ops
+    // (These ops require only bounded space, even with unbounded data)
     def map[A, B](ds: F[A])(f: A =>: B): F[B]
+    def flatMap[A, B](ds: F[A])(f: A =>: DataSet[B]): F[B]
```

Comment:
Do we want to force a `Dataset` here? I get that its a substitute for a "Finite set of data" but theoretically you can flatmap a stream as well - it just won't terminate unless its a finite stream.

Reply:
I'm actually against this, IMO neither, `DataSet` nor `DataStream` should be Monads, because that would mean you can no longer statically analyze them.

Reply:
In https://github.com/LukaJCB/analytics I used `concatMap` which uses `A => List[B]` :)